adapters/alkimi/alkimi.go

```diff
@@ -0,0 +1,194 @@
+			imp.BidFloor = bidFloorPrice.FloorMin
+		} else {
+			imp.BidFloor = extImpAlkimi.BidFloor
+		}
+		imp.Instl = extImpAlkimi.Instl
+		imp.Exp = extImpAlkimi.Exp
+
+		temp := extObj{AlkimiBidderExt: extImpAlkimi}
+		temp.AlkimiBidderExt.AdUnitCode = imp.ID
+
+		extJson, err := json.Marshal(temp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		imp.Ext = extJson
+		updatedImps = append(updatedImps, imp)
+	}
+
+	return updatedImps, errs
+}
+
+func buildBidderRequest(adapter *adapter, encoded []byte) *adapters.RequestData {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	reqBidder := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint,
+		Body:    encoded,
+		Headers: headers,
+	}
+	return reqBidder
+}
+
+// MakeBids will parse the bids from the Alkimi server
+func (adapter *adapter) MakeBids(request *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var errs []error
+
+	if adapters.IsResponseStatusCodeNoContent(response) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+
+	var bidResp openrtb2.BidResponse
+	err := json.Unmarshal(response.Body, &bidResp)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	bidCount := len(bidResp.SeatBid)
+	if bidCount == 0 {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "Empty SeatBid array",
+		}}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(bidCount)
+	for _, seatBid := range bidResp.SeatBid {
+		for _, bid := range seatBid.Bid {
+			copyBid := bid
+			resolveMacros(&copyBid)
+			impId := copyBid.ImpID
+			imp := request.Imp
+			bidType, err := getMediaTypeForImp(impId, imp)
+			if err != nil {
+				errs = append(errs, err)
+			}
+			bidderBid := &adapters.TypedBid{
+				Bid:     &copyBid,
+				BidType: bidType,
+			}
+			bidResponse.Bids = append(bidResponse.Bids, bidderBid)
+		}
+	}
+	return bidResponse, errs
+}
+
+func resolveMacros(bid *openrtb2.Bid) {
+	strPrice := strconv.FormatFloat(bid.Price, 'f', -1, 64)
+	bid.NURL = strings.Replace(bid.NURL, price_macro, strPrice, -1)
+	bid.AdM = strings.Replace(bid.AdM, price_macro, strPrice, -1)
+}
+
+func getMediaTypeForImp(impId string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Banner != nil {
+				return openrtb_ext.BidTypeBanner, nil
+			}
+			if imp.Video != nil {
+				return openrtb_ext.BidTypeVideo, nil
+			}
+			if imp.Audio != nil {
+				return openrtb_ext.BidTypeAudio, nil
+			}
+			if imp.Native != nil {
+				return openrtb_ext.BidTypeNative, nil
+			}
+		}
+	}
+	return openrtb_ext.BidTypeBanner, &errortypes.BadInput{
+		Message: fmt.Sprintf("Failed to find imp \"%s\"", impId),
+	}
```

Comment:
can return empty string instead of `openrtb_ext.BidTypeBanner`
```
return "", &errortypes.BadInput{
		Message: fmt.Sprintf("Failed to find imp \"%s\"", impId),
	}

```